src/main/kotlin/dev/minn/jda/ktx/InlineShardManagerBuilder.kt

```diff
@@ -0,0 +1,542 @@
+    builder: InlineShardManagerBuilder.() -> Unit
+                                   ) = InlineShardManagerBuilder(DefaultShardManagerBuilder.createLight(token, intents.toList()))
+        .also(builder)
+
+/**
+ * Creates a new [ShardManager] instance with the default settings by using [DefaultShardManagerBuilder.create]
+ *
+ * @param token The token to log in with. If set to null, assign a value to [InlineShardManagerBuilder.token].
+ * @param login Whether or not to automatically log in. See: [ShardManager.login]
+ * @param intents A list of intents to enable.
+ * @param builder The function used to configure the builder.
+ * @return The new [ShardManager] instance.
+ * @see DefaultShardManagerBuilder.create
+ */
+inline fun ShardManager(
+    token: String? = null,
+    login: Boolean = true,
+    vararg intents: GatewayIntent,
+    builder: InlineShardManagerBuilder.() -> Unit
+                       ) = InlineShardManagerBuilder(DefaultShardManagerBuilder.create(token, intents.toList()))
+        .also(builder)
+        .build(login)
+
+/**
+ * Creates a new [ShardManager] instance with the default settings by using [DefaultShardManagerBuilder.createLight]
+ *
+ * @param token The token to log in with. If set to null, assign a value to [InlineShardManagerBuilder.token].
+ * @param intents A list of intents to enable.
+ * @param builder The function used to configure the builder.
+ * @return The [ShardManager] instance. You must call [DefaultShardManagerBuilder.build] on this to finish constructing the ShardManager.
+ * @see DefaultShardManagerBuilder.createLight
+ */
+inline fun ShardManagerBuilder(
+    token: String? = null,
+    vararg intents: GatewayIntent,
+    builder: InlineShardManagerBuilder.() -> Unit
+                              ) = InlineShardManagerBuilder(DefaultShardManagerBuilder.create(token, intents.toList()))
+        .also(builder)
+
+/**
+ * Makes construction of a [DefaultShardManager] much more idiomatic and cleaner to do in kotlin.
+ *
+ * @property builder The delegate [DefaultShardManagerBuilder].
+ * @constructor Create an inline shard manager builder to make construction of the [DefaultShardManager] more idiomatic in kotlin.
+ */
+class InlineShardManagerBuilder(val builder: DefaultShardManagerBuilder) {
+    var rawEvents: Boolean = false
+        set(value) {
+            builder.setRawEventsEnabled(value)
+            field = value
+        }
+    
+    var relativeRateLimit: Boolean = true
+        set(value) {
+            builder.setRelativeRateLimit(value)
+            field = value
+        }
+    
+    var enableCache: Collection<CacheFlag> = emptySet()
+        set(value) {
+            builder.enableCache(value)
+            field = value
+        }
+    
+    var disableCache: Collection<CacheFlag> = emptySet()
+        set(value) {
+            builder.disableCache(value)
+            field = value
+        }
+    
+    var memberCachePolicy: MemberCachePolicy? = null
```

Comment:
I still think these should represent the correct defaults that are applied to the builders

Reply:
The only problem is, that the defaults within the builder are subject to change. There is no reason that they will remain the same across versions. And since the defaults are not accessible except via reflection, there is no nice way to guarantee parity with the JDA builder defaults.